taskRequests/script.js

```diff
@@ -0,0 +1,188 @@
+const API_BASE_URL = window.API_BASE_URL;
+const taskRequestContainer = document.getElementById('task-request-container');
+const containerBody = document.querySelector('.container__body');
+const filterContainer = document.querySelector('.container__filters');
+
+const fetchedTaskRequests = [];
+
+const loader = document.querySelector('.container__body__loader');
+const startLoading = () => loader.classList.remove('hidden');
+const stopLoading = () => loader.classList.add('hidden');
+
+function createCustomElement(domObjectMap) {
+  const el = document.createElement(domObjectMap.tagName);
+  for (const [key, value] of Object.entries(domObjectMap)) {
+    if (key === 'tagName') {
+      continue;
+    }
+    if (key === 'eventListeners') {
+      value.forEach((obj) => {
+        el.addEventListener(obj.event, obj.func);
+      });
+    }
+    if (key === 'class') {
+      if (Array.isArray(value)) {
+        el.classList.add(...value);
+      } else {
+        el.classList.add(value);
+      }
+    } else if (key === 'child') {
+      el.append(...value);
+    } else {
+      el[key] = value;
+    }
+  }
+  return el;
+}
+
+async function getTaskRequests() {
+  startLoading();
+  try {
+    const res = await fetch(`${API_BASE_URL}/taskRequests`);
+
```

Comment:
can we not create constants for the API and then use it, maybe this API gets used somewhere and in the future we decide to change the API and for that we need to go on each file and change it instead of that we make changes in one file

Reply:
addressed in the above comment